RustLeanModels/Basic.lean

```diff
@@ -52,44 +55,61 @@ theorem Eq_iff_of_unique_and_mp : unique p → (∀ y, (x = y) → p y) → (∀
   unfold unique at g1; exact g1 x y g2 g3
 
 @[simp]
-lemma not_sym :  ¬ x = y →  ¬ y = x := by intro g; by_contra; rename_i g1; symm at g1; contradiction
+theorem not_sym :  ¬ x = y →  ¬ y = x := by intro g; intro h; have h := Eq.symm h; contradiction
 
-lemma exists_EQ {α: Type} {p q : α → Prop}: (∀ x, p x  ↔ q x) → ((∃ y, p y) ↔ (∃ y, q y)) :=by
+theorem exists_EQ {α: Type} {p q : α → Prop}: (∀ x, p x  ↔ q x) → ((∃ y, p y) ↔ (∃ y, q y)) :=by
   intro g; constructor
-  intro gy ; obtain ⟨y,gy⟩ := gy ; rw[g y] at gy ; use y
-  intro gy ; obtain ⟨y,gy⟩ := gy ; rw[← g y] at gy ; use y
-
-lemma not_in_forall_not_eq {s: List α} (h: ∀ x ∈ s, a ≠ x) : a ∉ s :=by
-  induction s; simp; rename_i head tail ind
-  simp_all only [ne_eq, mem_cons, or_true, not_false_eq_true, implies_true, forall_const,
-    forall_true_left, forall_eq_or_imp, not_sym, or_self]
+  · intro gy ; obtain ⟨y,gy⟩ := gy ; rw[g y] at gy ; exists y
+  · intro gy ; obtain ⟨y,gy⟩ := gy ; rw[← g y] at gy ; exists y
 
+theorem not_in_forall_not_eq {s: List α} (h: ∀ x ∈ s, a ≠ x) : a ∉ s :=by
+  induction s
+  · simp
+  · rename_i head tail ind
+    simp_all only [ne_eq, mem_cons, or_true, not_false_eq_true, implies_true, forall_const,
+    forall_eq_or_imp, not_sym, or_self]
 
 -- NAT
-lemma cases_max : x = Nat.max m n → (x=m) ∨ (x=n) :=by
-  by_cases m ≤ n ; intro g; rename_i g0; simp only [g0, max_eq_right] at g; simp only [g, or_true]
-  intro g; rename_i g0; simp only [not_le] at g0; have g0:= le_of_lt g0;
-  simp only [g0, max_eq_left] at g; simp only [g, true_or]
-
-lemma max_add_right : Nat.max a (b+c) ≤ c + Nat.max a b :=by
-  by_cases a ≤ b ; rename_i g; simp only [g, max_eq_right, max_le_iff]; omega
-  rename_i g; simp only [not_le] at g; have g:= le_of_lt g;
-  simp only [g, max_eq_left, max_le_iff,le_add_iff_nonneg_left, _root_.zero_le, true_and, ge_iff_le]
+theorem cases_max : x = Nat.max m n → (x=m) ∨ (x=n) :=by
```

Comment:
We could  simplify this proof by using `Nat.max_def`:
```
theorem cases_max : x = Nat.max m n → x = m ∨ x = n := by
  rw [Nat.max, Nat.max_def]
  split <;> simp_all
```

Reply:
Done.